### PR TITLE
Remove `dual_screen` from `new_gallery` integration test

### PR DESCRIPTION
**Pull request by gmackall**
Removes the `dual_screen` package from `new_gallery`. Unblocks the fourth attempt to land https://github.com/flutter/engine/pull/53001. 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Review by jtmcdole — APPROVED** [review]

**Review by johnmccutchan — APPROVED** [review]

**Comment by gmackall** [comment]
Looks like we can't fake the tests running in pre-submit, ex:
https://ci.chromium.org/ui/p/flutter/builders/try/Linux_mokey%20new_gallery_impeller__transition_perf/1/overview

**Review by christopherfujino — APPROVED** [review]
looks really good to me

**Comment by gmackall** [comment]
Reason for revert: Causing the new_gallery tests to hang.

I can repro now though, so should be able to find a fix shortly

File: dev/integration_tests/new_gallery/lib/pages/home.dart [hunk 1]
```diff
@@ -1166,47 +1164,8 @@ class _StudyWrapperState extends State<StudyWrapper> {
               child: widget.study,
             ),
           ),
-          if (!isDisplayFoldable(context))
```
Comment (gmackall) [inline dev/integration_tests/new_gallery/lib/pages/home.dart:1169]: This is almost certainly the mistake - given that the conditional is `!isDisplayFoldable(context)` I shouldn't have deleted the contents, but rather left them around. Will test and put up a new pr if that's correct

**Pull request merged**